openeo-geotrellis/src/test/scala/org/openeo/geotrellis/collectiontests/CollectionTests.scala

```diff
@@ -0,0 +1,339 @@
+    val isEqual = jsonEquals(json1, json2)
+    assertTrue(isEqual)
+  }
+
+  @Test
+  def jsonEqualsMixedTest(): Unit = {
+    val mapper = new ObjectMapper()
+    val json1 = mapper.readTree("""{"a": 1, "b": {"c": 2.0}}""")
+    val json2 = mapper.readTree("""{"b": {"c": 2.0000002}, "a": 1.0000001}""")
+    val isEqual = jsonEquals(json1, json2)
+    assertTrue(isEqual)
+  }
+
+  @Test
+  def jsonEqualsListTest(): Unit = {
+    val mapper = new ObjectMapper()
+    val json1 = mapper.readTree("""[1, 3.1415]""")
+    val json2 = mapper.readTree("""[1, 3.1415926]""")
+    val isEqual = jsonEquals(json1, json2)
+    assertTrue(isEqual)
+  }
+
+  @Test
+  def jsonDifferentListTest(): Unit = {
+    val mapper = new ObjectMapper()
+    val json1 = mapper.readTree("""[1, 3.1415]""")
+    val json2 = mapper.readTree("""[5, 3.1415]""")
+    val isEqual = jsonEquals(json1, json2)
+    assertFalse(isEqual)
+  }
+
+  @Test
+  def jsonDifferentLengthListTest(): Unit = {
+    val mapper = new ObjectMapper()
+    val json1 = mapper.readTree("""[1, 3.1415]""")
+    val json2 = mapper.readTree("""[1, 3.1415, 1]""")
+    val isEqual = jsonEquals(json1, json2)
+    assertFalse(isEqual)
+  }
+
+  @Test
+  def jsonNotEqualsTest(): Unit = {
+    val mapper = new ObjectMapper()
+    val json1 = mapper.readTree("""{"a": 1, "b": {"c": 2.0}}""")
+    val json2 = mapper.readTree("""{"a": null, "b": {"c": 2.0000002}}""")
+    val isEqual = jsonEquals(json1, json2)
+    println(isEqual)
+    assertFalse(isEqual)
+  }
+
+  @ParameterizedTest
+  @MethodSource(Array("testLayerParams"))
+  def testLayers(layerStr: String): Unit = {
+    testLayerImpl(layerStr)
+  }
+
+  /**
+   * This functin shares the same structure as the one of openeo-collection-tests,
+   * But for the moment has different expected results.
+   */
+  private def testLayerImpl(layerStr: String,
+                            input_file: Option[String] = None,
+                            from_date: String = "2020-03-01T00:00:00Z",
+                            to_date: String = "2020-03-31T00:00:00Z",
+                            output_dir: Option[String] = None,
+                            expected_dir: Option[String] = None,
+                           ): Unit = {
+    val output_dir_get = output_dir.getOrElse(new java.io.File("./tmp_collectiontests/").getCanonicalPath)
+    Files.createDirectories(Paths.get(output_dir_get))
+
+    val expected_dir_get = expected_dir.getOrElse(input_file.getOrElse(getClass.getResource("/org/openeo/geotrellis/collectiontests/expected/").getPath))
+
+    val vector_file = input_file.getOrElse(getClass.getResource("/org/openeo/geotrellis/collectiontests/"
+      + (if (layerStr.contains("CGLS")) "cgls_test.json" else "50testfields.json")).getFile)
+    var polygons = ProjectedPolygons.fromVectorFile(vector_file)
+    polygons = ProjectedPolygons.reproject(polygons, WebMercator) // LatLng gives slightly different rsults. Still not the same as the Python implmentation
```

Comment:
the crs to use for sentinel layers is EPSG:32631, webmercator will give non-exact results

Reply:
Using EPSG:32631 now.